### PR TITLE
Add documentation for connecting oidc identities to django users

### LINKED ISSUE #88: document how auth0 identities match to django identities
There's no documentation covering how mozilla-django-oidc takes an auth0 identity from a user_token and figures out which Django user that belongs to.

It's pretty important for any site that has to migrate from an existing system to this one. Further, surfacing how that works will likely surface inherent restrictions regarding accounts.

### PR DESCRIPTION
**Pull request by willkg**
Fixes #88.

### CODE REVIEW

**Comment by willkg** [comment]
Does this look ok? What can I/we improve?

File: docs/installation.rst [hunk 1]
```diff
@@ -76,3 +76,48 @@ Please add the following to your ``settings.py``:
...
+For example, let's say we store the email address in a ``Profile`` table
+in a field that's marked unique so multiple users can't have the same
+email address. Then we could do this:
+
+.. code-block:: python
+
+   from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+
+   class MyOIDCAB(OIDCAuthenticationBackend):
+       def filter_users_by_claims(self, claim):
```
Comment (willkg) [inline docs/installation.rst]: Part of me was wondering whether we should talk about what's in the claim and/or tell the user/developer how to figure out what's in the claim. I'm pretty sure the information in the claim is different depending on whether you're using Mozilla's SSO or Auth0 or some other OIDC provider, so I'm not sure how to do this effectively. Maybe have a paragraph after this along the lines of:

> The information available in the claim depends on which OIDC provider you're using. Best way to figure out what's in the claim is to log the information in a development environment.

Maybe we should also add?:

> The data in the claim isn't static and can change over time and also be different depending on the user that logged in.

Are both of those true in your experience? Should we add them to reduce the surprises users/developers might encounter when trying to integrate mozilla-django-oidc into their sites?

Comment (akatsoulas) [inline docs/installation.rst]: I believe both these paragraphs will help users/developers to better understand what to expect regarding the data in the claim. 👍 for adding those.

File: docs/installation.rst [hunk 2]
```diff
@@ -76,3 +76,48 @@ Please add the following to your ``settings.py``:
...
+Connecting OIDC user identities to Django users
+-----------------------------------------------
+
+By default, mozilla-django-oidc looks up a Django user matching the email field
+to the email address returned in the user info data from the OIDC provider.
+
+This means that no two users in the Django user table can have the same email
+address. Since the email field is not unique, it's possible that this can
+happen. Especially if you allow users to change their email address. If it ever
+happens, then the users in question won't be able to authenticate.
```
Comment (akatsoulas) [inline docs/installation.rst]: Does it make sens to add here a couple of lines about the `default_username_algo` and how it is used to populate the django username?

Comment (willkg) [inline docs/installation.rst]: I think it does, too. I was writing a section about Django user creation, but the API looked weird and so I wrote up #91. I figure when I fix that issue, I'll add some documentation about it, too.

**Comment by akatsoulas** [comment]
@willkg is this ready for merge?

**Comment by willkg** [comment]
@akatsoulas Yes, please!

I'm trying to keep PRs small and topical to make them easier to work through. I can make bigger PRs if that's easier. Just let me know.

**Comment by akatsoulas** [comment]
No need for bigger PRs. I was just wondering for the current one because of the two paragraphs in the code block in your comment. Thanks!

**Pull request merged**